libs/langchain/langchain/vectorstores/redis/base.py

```diff
@@ -0,0 +1,1351 @@
+                from Redis.
+
+        Returns:
+            Dict[str, Any]: Collected metadata.
+        """
+        # new metadata dict as modified by this method
+        meta = {}
+        for key in self._schema.metadata_keys:
+            try:
+                meta[key] = getattr(result, key)
+            except AttributeError:
+                # warning about attribute missing
+                logger.warning(
+                    f"Metadata key {key} not found in metadata. "
+                    + "Setting to None. \n"
+                    + "Metadata fields defined for this instance: "
+                    + f"{self._schema.metadata_keys}"
+                )
+                meta[key] = None
+        return meta
+
+    def _prepare_query(
+        self,
+        query: str,
+        k: int = 4,
+        filter: Optional[RedisFilterExpression] = None,
+        distance_threshold: Optional[float] = None,
+        with_metadata: bool = True,
+        with_distance: bool = False,
+    ) -> Tuple["Query", Dict[str, Any]]:
+        # Creates embedding vector from user query
+        embedding = self.embedding_function(query)
+
+        # Creates Redis query
+        params_dict: Dict[str, Union[str, bytes, float]] = {
+            "vector": _array_to_buffer(embedding, self._schema.vector_dtype),
+        }
+
+        # prepare return fields including score
+        return_fields = [self._schema.content_key]
+        if with_distance:
+            return_fields.append("distance")
+        if with_metadata:
+            return_fields.extend(self._schema.metadata_keys)
+
+        if distance_threshold:
+            params_dict["distance_threshold"] = distance_threshold
+            return (
+                self._prepare_range_query(
+                    k, filter=filter, return_fields=return_fields
+                ),
+                params_dict,
+            )
+        return (
+            self._prepare_vector_query(k, filter=filter, return_fields=return_fields),
+            params_dict,
+        )
+
+    def _prepare_range_query(
+        self,
+        k: int,
+        filter: Optional[RedisFilterExpression] = None,
+        return_fields: List[str] = [],
+    ) -> "Query":
+        try:
+            from redis.commands.search.query import Query
+        except ImportError as e:
+            raise ImportError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
+            ) from e
+        vector_key = self._schema.content_vector_key
+        base_query = f"@{vector_key}:[VECTOR_RANGE $distance_threshold $vector]"
+
+        if filter:
+            base_query = "(" + base_query + " " + str(filter) + ")"
+
+        query_string = base_query + "=>{$yield_distance_as: distance}"
+
+        return (
+            Query(query_string)
+            .return_fields(*return_fields)
+            .sort_by("distance")
+            .paging(0, k)
```

Comment:
One of the advantages of range queries is that you can get all the results that are within a given range (not just the top k ones), and in particular you can see how many results are within the range. To utilize this feature I would suggest having k as an optional argument (and perhaps using an upper bound such as 1000 for paging), if that makes sense to you.

Reply:
agreed -- was going to suggest using K as an optional upper bound. @alonre24 isn't paging a default param 0-10 though?

Maybe something like:
```python
query = (
    Query(query_string)
    .return_fields(*return_fields)
    .sort_by("distance")
    .dialect(2)
)

if k:
    query = query.paging(0, k)
```

Reply:
Yes, the default paging is 10, that's why I suggest also choosing an upper bound, so we have: 
```py
query = query.paging(0, k if k else UPPER_BOUND)
```

Reply:
It's part of the higher level interface abstraction so it would have to be set. I've started this conversation the lc folks though. Most likely the best route is another method that exposes this feature better.